JamfSync/Model/Jcds2Dp.swift

```diff
@@ -193,40 +201,419 @@ class Jcds2Dp: DistributionPoint {
+        
+        request.addValue("AWS4-HMAC-SHA256 Credential=\(accessKeyId)/\(dateString.prefix(8))/\(region)/s3/aws4_request,SignedHeaders=date;host;x-amz-content-sha256;x-amz-date;x-amz-security-token,Signature=\(signatureProvided)", forHTTPHeaderField: "Authorization")
+        
+        URLCache.shared.removeAllCachedResponses()
+        
+        do {
+            let (responseData, response) = try await urlSession.upload(for: request, from: chunk)
+            
+            let responseString = String(data: responseData, encoding: .utf8) ?? ""
+            
+            let httpResponse = response as? HTTPURLResponse
+            let allHeaders = httpResponse?.allHeaderFields
+            
+            print("[multipartUpload] partNumber: \(partNumber) - Etag: \(allHeaders?["Etag"] ?? "")")
+            partNumberEtagList.append(CompleteMultipart(partNumber: partNumber, eTag: "\(allHeaders?["Etag"] ?? "")"))
+
+            return(.success(()))
+        } catch {
+            return(.failure(error))
+        }
+    }
+    
+    private func completeMultipartUpload(fileUrl: URL, completeMultipartUploadXml: String, uploadId: String) async -> String {
+                
+        let packageToUpload = fileUrl.lastPathComponent
+        
+        var urlHostAllowedPlus = CharacterSet.urlHostAllowed
+        urlHostAllowedPlus.remove(charactersIn: "+")
+        let encodedPackageName = packageToUpload.addingPercentEncoding(withAllowedCharacters: urlHostAllowedPlus) ?? ""
+        
+        let bucket          = initiateUploadData?.bucketName ?? ""
+        let region          = initiateUploadData?.region ?? ""
+        let key             = (initiateUploadData?.path ?? "") + encodedPackageName
+        let accessKeyId     = initiateUploadData?.accessKeyID ?? ""
+        let secretAccessKey = initiateUploadData?.secretAccessKey ?? ""
+        let sessionToken    = initiateUploadData?.sessionToken ?? ""
+        let contentType     = ""
+        let jcdsServerURL   = ( region == "us-east-1" ) ? URL(string: "https://\(bucket).s3.amazonaws.com/\(key)" + "?uploadId=\(uploadId)")!:URL(string: "https://\(bucket).s3-\(region).amazonaws.com/\(key)" + "?uploadId=\(uploadId)")!
+        
+        let currentDate = Date()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyyMMdd'T'HHmmss'Z'"
+        dateFormatter.timeZone = TimeZone(identifier: "UTC")
+        let dateString = dateFormatter.string(from: currentDate)
+        
+        var request = URLRequest(url: jcdsServerURL,
+                                 cachePolicy: .reloadIgnoringLocalCacheData)
+        
+        
+        request.addValue("\(currentDate)", forHTTPHeaderField: "Date")
```

Comment:
This request is very similar to the one in createMultipartUpload so it would be good if they could be consolidated into one function.